executor/src/witgen/machines/fixed_lookup_machine.rs

```diff
@@ -279,14 +284,43 @@ impl<T: FieldElement> FixedLookup<T> {
 
     fn process_range_check<'b>(
         &self,
+        rows: &RowPair<'_, '_, T>,
         lhs: &AffineExpression<&'b PolynomialReference, T>,
         rhs: &'b PolynomialReference,
     ) -> EvalResult<'b, T> {
+        // Use AffineExpression::solve_with_range_constraints to transfer range constraints
+        // from the rhs to the lhs.
         let equation = lhs.clone() - AffineExpression::from_variable_id(rhs);
-        // TODO this is kind of unusual because we use symbolic fixed columns.
-        // we should better filter out the fixed column from the result.
-        let r = equation.solve_with_range_constraints(&self.global_constraints);
-        println!("{r:?}");
-        r
+        let range_constraints = UnifiedRangeConstraints {
+            witness_constraints: rows,
+            global_constraints: &self.global_constraints,
+        };
+        let updates = equation.solve_with_range_constraints(&range_constraints)?;
+
+        // Filter out any updates to the fixed columns and mark the identity as complete
+        Ok(EvalValue::complete(updates.constraints.into_iter().filter(
+            |(poly, _)| poly.poly_id().ptype == PolynomialType::Committed,
+        )))
+    }
+}
+
+/// Combines witness constraints on a concrete row with global range constraints
+/// (used for fixed columns).
+/// This is useful in order to transfer range constraints from fixed columns to
+/// witness columns (see [FixedLookup::process_range_check]).
+pub struct UnifiedRangeConstraints<'a, T: FieldElement> {
```

Comment:
Is this useful somewhere else as well?

Reply:
I thought so initially, but actually no. As you pointed out, fixed columns would normally be evaluated, so the range constraints are not relevant. It is only useful to transfer range constraints from fixed to witness columns, which is only done by this machine.